spring-cloud-starter-single-step-batch-job/src/main/java/org/springframework/cloud/task/batch/autoconfigure/kafka/KafkaItemReaderAutoConfiguration.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2020-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.task.batch.autoconfigure.kafka;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Properties;
+
+import org.springframework.batch.item.kafka.KafkaItemReader;
+import org.springframework.batch.item.kafka.builder.KafkaItemReaderBuilder;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.AutoConfigureAfter;
+import org.springframework.boot.autoconfigure.batch.BatchAutoConfiguration;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.autoconfigure.kafka.KafkaProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.util.StringUtils;
+
+/**
+ *
+ * Autconfiguration for a {@code KafkaItemReader}.
+ *
+ * @author Glenn Renfro
+ * @since 2.3
+ */
+@Configuration
+@EnableConfigurationProperties({ KafkaProperties.class, KafkaItemReaderProperties.class })
+@AutoConfigureAfter(BatchAutoConfiguration.class)
+public class KafkaItemReaderAutoConfiguration {
+
+	@Autowired
+	private KafkaProperties kafkaProperties;
+
+	@Bean
+	@ConditionalOnMissingBean
+	@ConditionalOnProperty(prefix = "spring.batch.job.kafkaitemreader", name = "name")
+	public KafkaItemReader<Object, Map<Object, Object>> kafkaItemReader(
+			KafkaItemReaderProperties kafkaItemReaderProperties) {
+		Properties consumerProperties = new Properties();
+		consumerProperties.putAll(this.kafkaProperties.getConsumer().buildProperties());
+		validateProperties(kafkaItemReaderProperties);
+		if (kafkaItemReaderProperties.getPartitions() == null
+				|| kafkaItemReaderProperties.getPartitions().size() == 0) {
+			kafkaItemReaderProperties.setPartitions(new ArrayList<>(1));
+			kafkaItemReaderProperties.getPartitions().add(0);
+		}
+		return new KafkaItemReaderBuilder<Object, Map<Object, Object>>()
+				.partitions(kafkaItemReaderProperties.getPartitions())
+				.consumerProperties(consumerProperties)
```

Comment:
I see `consumerProperties` are initialized from `this.kafkaProperties.getConsumer().buildProperties()` then passed to the reader. Please note that Spring Batch uses kafka constants like `ConsumerConfig.GROUP_ID_CONFIG`, `ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG`, etc for properties names which resolve respectively to `group.id`, `value.deserializer`, etc. Looking at the tests (and if I'm not mistaken), properties coming from `this.kafkaProperties.getConsumer().buildProperties()` will have a prefix `spring.kafka.consumer.` like `spring.kafka.consumer.group-id=1` and `"spring.kafka.consumer.value-deserializer="+ JsonDeserializer.class.getName()` . Those will not be accepted by the `KafkaItemReader`'s constructor or its builder.

Reply:
Hello @benas ,
In this case the kafkaProperties.getConsumer.BuildProperties sets the properties with the appropriate prefix.  As shown here: https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaProperties.java#L420

Now another question.   If users are used to using the ConsumerProperties prefix instead of the Spring Kafka prefix(ala boot style properties), we may want to support those here as well (as overrides).   Thoughts?

Reply:
If these single-step job starters are intended for boot users primarily, I would use the boot style first.